server/command.go

```diff
@@ -126,28 +141,50 @@ func (p *Plugin) settingsError(userID string, channelID string, errorText string
 }
 
 func (p *Plugin) executeSettingsCommand(c *plugin.Context, args *model.CommandArgs, parameters []string) (*model.CommandResponse, *model.AppError) {
+	l := p.getUserLocalizer(args.UserId)
 	text := ""
 
 	userConfig, err := p.getUserConfig(args.UserId)
 	if err != nil {
 		mlog.Debug("Unable to get user config", mlog.Err(err))
-		return p.settingsError(args.UserId, args.ChannelId, "Unable to get user settings.")
+		return p.settingsError(args.UserId, args.ChannelId, l.MustLocalize(&i18n.LocalizeConfig{
+			DefaultMessage: &i18n.Message{
+				ID:    "jitsi.command.settings.unable_to_get",
+				Other: "Unable to get user settings",
+			},
+		}))
 	}
 
 	if len(parameters) == 0 {
-		text = fmt.Sprintf("###### Jitsi Settings:\n* Embedded: `%v`\n* Naming Scheme: `%s`", userConfig.Embedded, userConfig.NamingScheme)
+		text = l.MustLocalize(&i18n.LocalizeConfig{
```

Comment:
I wonder if this should be synthesized in one function call. Building every time the `i18n.LocalizeConfig`, the `i18n.Message` looks very bulky. Maybe something like `l.GetLocalized(id, default, templateData)` would make things more legible.

Reply:
At least, we always need to use the i18n.Message struct, because is the one used by the `goi18n extract` command to extract the messages. But yes, I guess we can simplify it. Let me try.

Reply:
The main difference there, because we still need to pass the localizer and the params, and we need to use the `i18n.Message`, is a bit less verbose about the "fields" but I think that "verbose" approach is valuable, because you don't need to guess what is what.

The example converting it into a function:
```go
                 meetingTopic = p.localize(
                     l,
                     &i18n.Message{
                         ID:    "jitsi.start_meeting.personal_meeting_topic",
                         Other: "{{.Name}}'s Personal Meeting",
                     },
                     map[string]string{"Name": user.GetDisplayName(model.SHOW_NICKNAME_FULLNAME)},
                 )
```

Before was:
```go
                meetingTopic = l.MustLocalize(&i18n.LocalizeConfig{
                    DefaultMessage: &i18n.Message{
                        ID:    "jitsi.start_meeting.personal_meeting_topic",
                        Other: "{{.Name}}'s Personal Meeting",
                    },
                    TemplateData: map[string]string{"Name": user.GetDisplayName(model.SHOW_NICKNAME_FULLNAME)},
                })
```

I think i prefer the explicitness of the old approach, so I think I going to keep it as is now.

Reply:
I was thinking more on something like:
```
meetingTopic = l.localize("jitsi.start_meeting.personal_meeting_topic", "{{.Name}}'s Personal Meeting",
    map[string]string{"Name": user.GetDisplayName(model.SHOW_NICKNAME_FULLNAME)})
```

The implementation of l.localize would be something like:
```
func (l ...) Localize(ID string, default string, templateData map[string]string) {
    return l.MustLocalize(&i18n.LocalizeConfig{
                    DefaultMessage: &i18n.Message{
                        ID:    "jitsi.start_meeting.personal_meeting_topic",
                        Other: "{{.Name}}'s Personal Meeting",
                    },
                    TemplateData: map[string]string{"Name": user.GetDisplayName(model.SHOW_NICKNAME_FULLNAME)},
                }
}
```

Reply:
The problem with that is you can't extract automatically the strings with the `goi18n extract` because it is no longer a `i18n.Message` object with literal strings.

Reply:
Understood. Then it makes sense to leave it like this.